### PR TITLE
Some tests misusing assertTrue for comparisons fix

### LINKED ISSUE #16770: Some tests misusing assertTrue for comparisons
`assertTrue` is not for comparing arguments, should use `assertEqual` for that.

The developer's intent of the test was to compare argument 1 with argument 2, which is not happening. Really what is happening is the test is passing because first argument is truthy. The correct method to use is `assertEqual`

https://github.com/huggingface/transformers/blob/main/tests/wav2vec2/test_modeling_flax_wav2vec2.py#L389
https://github.com/huggingface/transformers/blob/main/tests/wav2vec2/test_modeling_flax_wav2vec2.py#L431
https://github.com/huggingface/transformers/blob/main/tests/wav2vec2/test_modeling_wav2vec2.py#L1064
https://github.com/huggingface/transformers/blob/main/tests/wav2vec2/test_modeling_wav2vec2.py#L1101
https://github.com/huggingface/transformers/blob/main/tests/wav2vec2/test_tokenization_wav2vec2.py#L205
https://github.com/huggingface/transformers/blob/main/tests/wav2vec2/test_tokenization_wav2vec2.py#L216
https://github.com/huggingface/transformers/blob/main/tests/wav2vec2/test_tokenization_wav2vec2.py#L217
https://github.com/huggingface/transformers/blob/main/tests/wav2vec2/test_tokenization_wav2vec2.py#L492
https://github.com/huggingface/transformers/blob/main/tests/wav2vec2/test_tokenization_wav2vec2.py#L499
https://github.com/huggingface/transformers/blob/main/tests/wav2vec2/test_tokenization_wav2vec2.py#L506
https://github.com/huggingface/transformers/blob/main/tests/wav2vec2_with_lm/test_processor_wav2vec2_with_lm.py#L371
https://github.com/huggingface/transformers/blob/main/tests/wav2vec2_with_lm/test_processor_wav2vec2_with_lm.py#L388
https://github.com/huggingface/transformers/blob/main/tests/test_sequence_feature_extraction_common.py#L188
https://github.com/huggingface/transformers/blob/main/tests/trainer/test_trainer_utils.py#L100
https://github.com/huggingface/transformers/blob/main/tests/trainer/test_trainer_utils.py#L102
https://github.com/huggingface/transformers/blob/main/tests/longformer/test_modeling_tf_longformer.py#L416
https://github.com/huggingface/transformers/blob/main/tests/longformer/test_modeling_tf_longformer.py#L489
https://github.com/huggingface/transformers/blob/main/tests/longformer/test_modeling_tf_longformer.py#L526
https://github.com/huggingface/transformers/blob/main/tests/longformer/test_modeling_longformer.py#L419
https://github.com/huggingface/transformers/blob/main/tests/longformer/test_modeling_longformer.py#L423
https://github.com/huggingface/transformers/blob/main/tests/longformer/test_modeling_longformer.py#L448
https://github.com/huggingface/transformers/blob/main/tests/longformer/test_modeling_longformer.py#L496
https://github.com/huggingface/transformers/blob/main/tests/longformer/test_modeling_longformer.py#L534
https://github.com/huggingface/transformers/blob/main/tests/trainer/test_trainer.py#L812
https://github.com/huggingface/transformers/blob/main/tests/trainer/test_trainer.py#L822
https://github.com/huggingface/transformers/blob/main/tests/wav2vec2_phoneme/test_tokenization_wav2vec2_phoneme.py#L268

I found this issue automatically, see other issues [here](https://codereview.doctor/huggingface/transformers)

### PR DESCRIPTION
**Pull request by code-review-doctor**
Fixes #16770

### CODE REVIEW

**Comment by code-review-doctor** [comment]
tests are failing now because the comparisons are actually being performed by the tests. Before the tests were just checking if the first argument was truthy, because that's what `assertTrue` does.

I'm just a static analysis bot (boop beep) so I'm not sure if the comparisons in the tests are wrong and therefore the tests need updating, or if the logic under test need fixing. Can someone please confirm.

**Comment by patrickvonplaten** [comment]
Hey @code-review-doctor,

It seems like you uncovered some bugs in the testing - thanks a lot! Do you want to fix them or should I go ahead and fix them?

**Comment by code-review-doctor** [comment]
@patrickvonplaten glad to be of service :)

If you could go ahead and fix them that would be great. My "thing" is running static analysis checkers on hundreds of open source projects and then letting them know so they can fixed properly :)

FYI and FWIW I have [GitHub integration](https://github.com/marketplace/django-doctor/) so I can review your PRs to prevent problems like this in future :)

File: tests/longformer/test_modeling_longformer.py [hunk 1]
```diff
@@ -416,11 +416,11 @@ def test_diagonalize(self):
 
     def test_pad_and_transpose_last_two_dims(self):
         hidden_states = self._get_hidden_states()
-        self.assertTrue(hidden_states.shape, (1, 8, 4))
+        self.assertEqual(hidden_states.shape, (1, 4, 8))
```
Comment (patrickvonplaten) [inline tests/longformer/test_modeling_longformer.py:419]: This was recently correct by @ydshieh if I remember correctly

**Review by sgugger — APPROVED** [review]
Thanks a lot for fixing all those tests!

**Comment by patrickvonplaten** [comment]
Thanks a lot @code-review-doctor !

File: tests/longformer/test_modeling_longformer.py (hunk 1 above)
Comment (ydshieh) [reply tests/longformer/test_modeling_longformer.py:419]: **[Just to add some context]**

The PR #15537 didn't touch the details like `LongformerSelfAttention._pad_and_transpose_last_two_dims`.

In the simplest words, that PR just performed `unpadding` in the final outputs in `(TF)LongformerEncoder`.

----------------------------
Thanks again, @code-review-doctor !

**Pull request merged**